### PR TITLE
Fix: Add --dry-run to the tool for applying state-parts

### PR DESCRIPTION
**Pull request by nikurt**
Also rework the state parts command API to be a single command with two subcommands instead of two separate commands.

### CODE REVIEW

**Comment by akhi3030** [comment]
I think @marcelo-gonzalez might be a good reviewer for this PR.  As they are not a codeowner, I am happy to approve once they do the review.

**Review by marcelo-gonzalez — APPROVED** [review]

File: tools/state-viewer/src/state_parts.rs
Comment (nikurt) [reply tools/state-viewer/src/state_parts.rs:84]: Agreed. Done.

File: tools/state-viewer/src/cli.rs
Comment (nikurt) [reply tools/state-viewer/src/cli.rs:31]: Mentioned in the `CHANGELOG`.
I can only imagine StateParts and StateChanges commands to be useful to somebody working on State Sync. Very unlikely to break anyone.

File: tools/state-viewer/src/state_parts.rs
Comment (nikurt) [reply tools/state-viewer/src/state_parts.rs:36]: Would be nice, but `clap` doesn't allow more than 1 subcommand per command.
Have to go with a common sub-sub-command.

**Review by akhi3030 — APPROVED** [review]

File: tools/state-viewer/src/state_parts.rs
Comment (marcelo-gonzalez) [reply tools/state-viewer/src/state_parts.rs:38]: ah no i meant like as one of the options in `struct StatePartsCmd`. not a huge deal though obviously

**Pull request merged**
